### PR TITLE
Refactor the state machines, expand on blade state machine

### PR DESCRIPTION
**Pull request by Jim3Things**
Not ready for check in - this for an early look

### CODE REVIEW

File: internal/services/inventory/messages/SimControl.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+// StartSim is the message used to signal the start of a rack-level simulation.
+type StartSim struct {
+	EnvelopeState
+}
+
+func NewStartSim(ctx context.Context, ch chan *sm.Response) *StartSim {
+	msg := &StartSim{}
+	msg.Initialize(ctx, ch)
+	msg.Tag = TagStartSim
```
Comment (MikeCJohnsonUk) [inline internal/services/inventory/messages/SimControl.go:17]: Should perhaps this be a parameter on the Initialize() ?

File: internal/services/inventory/blade.go [hunk 2]
```diff
@@ -144,13 +164,124 @@ func newBlade(def *pbc.BladeCapacity, r *rack, id int64) *blade {
...
-		sm.WithFirstState(bladeOffState, &bladeOff{}),
-		sm.WithState(bladePoweredState, &bladePowered{}),
-		sm.WithState(bladeBootingState, &bladeBooting{}),
-		sm.WithState(bladeWorkingState, &bladeWorking{}),
-		sm.WithState(bladeWorkloadStoppingState, &bladeWorkloadStopping{}),
-		sm.WithState(bladeStoppingState, &bladeStopping{}),
-		sm.WithState(bladeFaultedState, &bladeFaulted{}))
+		sm.WithFirstState(
+			bladeStart,
+			"start",
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply internal/services/inventory/blade.go]: I've changed the SM to use the string as the identifying key and removed the index.

File: internal/services/inventory/messages/SimControl.go (hunk 1 above)
Comment (Jim3Things) [reply internal/services/inventory/messages/SimControl.go]: done.

File: internal/services/inventory/tor.go [hunk 3]
```diff
@@ -137,7 +135,7 @@ func workingSetConnection(ctx context.Context, machine *sm.SimpleSM, m sm.Envelo
 		c, ok = t.cables[id]
 	}
 
-	if !ok {
+	if c == nil || !ok {
```
[the remark this answers is not in the document]
Comment (MikeCJohnsonUk) [reply internal/services/inventory/tor.go:138]: Oh, it's parsing a message. Ok, makes sense now.

**Review by MikeCJohnsonUk — APPROVED** [review]

**Pull request merged**